### PR TITLE
Parity Ethereum 2.0.0

### PR DESCRIPTION
**Pull request by 5chdn**
- rename `parity` ~~crate~~ to `parity-ethereum`
- bump version to 2.0.0

### CODE REVIEW

**Comment by debris** [comment]
does`2.0.0` mean that we can break backwards compatibility? 🤔

**Comment by 5chdn** [comment]
> does `2.0.0` mean that we can break backwards compatibility?

hehe, go ahead.

File: ethcore/sync/src/chain/propagator.rs [hunk 1]
```diff
@@ -48,7 +48,7 @@ fn accepts_service_transaction(client_id: &str) -> bool {
 	// Parity versions starting from this will accept service-transactions
 	const SERVICE_TRANSACTIONS_VERSION: (u32, u32) = (1u32, 6u32);
 	// Parity client string prefix
-	const PARITY_CLIENT_ID_PREFIX: &'static str = "Parity/v";
+	const PARITY_CLIENT_ID_PREFIX: &'static str = "Parity-Ethereum/v";
 
 	if !client_id.starts_with(PARITY_CLIENT_ID_PREFIX) {
```
Comment (5chdn) [inline ethcore/sync/src/chain/propagator.rs]: Are we seriously checking compatibility via version string? If yes, do you want me to check for both?

```
const LEGACY_CLIENT_ID_PREFIX: &'static str = "Parity/v";
const PARITY_CLIENT_ID_PREFIX: &'static str = "Parity-Ethereum/v";
if !client_id.starts_with(LEGACY_CLIENT_ID_PREFIX) && !client_id.starts_with(PARITY_CLIENT_ID_PREFIX) {
    // ...
}
```

File: ethcore/sync/src/chain/propagator.rs [hunk 2]
```diff
@@ -48,16 +48,24 @@ fn accepts_service_transaction(client_id: &str) -> bool {
...
 	// Parity client string prefix
-	const PARITY_CLIENT_ID_PREFIX: &'static str = "Parity/v";
+	const LEGACY_CLIENT_ID_PREFIX: &'static str = "Parity/v";
+	const PARITY_CLIENT_ID_PREFIX: &'static str = "Parity-Ethereum/v";
 
-	if !client_id.starts_with(PARITY_CLIENT_ID_PREFIX) {
-		return false;
-	}
+	if client_id.starts_with(LEGACY_CLIENT_ID_PREFIX) {
+	let ver: Vec<u32> = client_id[LEGACY_CLIENT_ID_PREFIX.len()..].split('.')
```
Comment (dvdplm) [inline ethcore/sync/src/chain/propagator.rs]: This line should be indented (and the 4 following lines as well).

File: ethcore/sync/src/chain/propagator.rs [hunk 3]
```diff
@@ -48,16 +48,24 @@ fn accepts_service_transaction(client_id: &str) -> bool {
...
+		.take(2)
+		.filter_map(|s| s.parse().ok())
+		.collect();
+		ver.len() == 2 && (ver[0] > SERVICE_TRANSACTIONS_VERSION.0 || (ver[0] == SERVICE_TRANSACTIONS_VERSION.0 && ver[1] >= SERVICE_TRANSACTIONS_VERSION.1))
+	} else if client_id.starts_with(PARITY_CLIENT_ID_PREFIX) {
 	let ver: Vec<u32> = client_id[PARITY_CLIENT_ID_PREFIX.len()..].split('.')
```
Comment (dvdplm) [inline ethcore/sync/src/chain/propagator.rs]: indentation like above

File: ethcore/sync/src/chain/propagator.rs [hunk 4]
```diff
@@ -48,16 +48,24 @@ fn accepts_service_transaction(client_id: &str) -> bool {
...
 
-	if !client_id.starts_with(PARITY_CLIENT_ID_PREFIX) {
+	if client_id.starts_with(LEGACY_CLIENT_ID_PREFIX) {
+		let ver: Vec<u32> = client_id[LEGACY_CLIENT_ID_PREFIX.len()..].split('.')
+			.take(2)
+			.filter_map(|s| s.parse().ok())
+			.collect();
+			ver.len() == 2 && (ver[0] > SERVICE_TRANSACTIONS_VERSION.0 || (ver[0] == SERVICE_TRANSACTIONS_VERSION.0 && ver[1] >= SERVICE_TRANSACTIONS_VERSION.1))
+	} else if client_id.starts_with(PARITY_CLIENT_ID_PREFIX) {
+		let ver: Vec<u32> = client_id[PARITY_CLIENT_ID_PREFIX.len()..].split('.')
```
Comment (ngotchac) [inline ethcore/sync/src/chain/propagator.rs]: Would be cleaner to do the split in the ifs, and then the rest outside : 
```
let split = if ... { ...split() } else if { ...split() } else { return false };
let ver: Vec<u32> = ...
```

**Comment by 5chdn** [comment]
Ok, this finally works.

**Pull request merged**